### PR TITLE
Task 9359: wallet screen - fetch data and api

### PR DESCRIPTION
**Pull request by vanpd-0859**
### Issue Link :link:

https://edu-redmine.sun-asterisk.vn/issues/9359
### Goals :soccer:

Fetch data from API and display on UI:
- Total value charts
- List assets, its information

### CODE REVIEW

**Comment by vanpd-0859** [comment]
ready

File: CryptoWallet/Repositories/EthereumInteraction.swift [hunk 1]
```diff
@@ -82,4 +84,35 @@ enum EthereumInteraction {
                                     isHierarchicalDeterministic: true)
         return importedWallet
     }
+    
+    static func getERC20TokenBalance(contractAddress: String, walletAddress: String) -> String? {
+        let web3 = Web3.InfuraMainnetWeb3()
+        let _walletAddress = EthereumAddress(walletAddress)
```
Comment (tienpx3010) [inline CryptoWallet/Repositories/EthereumInteraction.swift]:
```suggestion
        let walletAddress = EthereumAddress(walletAddress)
```

File: CryptoWallet/Repositories/EthereumInteraction.swift [hunk 2]
```diff
@@ -82,4 +84,35 @@ enum EthereumInteraction {
...
+        let erc20ContractAddress = EthereumAddress(contractAddress)
+        guard let contract = web3.contract(Web3.Utils.erc20ABI, at: erc20ContractAddress, abiVersion: 2) else {
+            return nil
+        }
+        var options = TransactionOptions.defaultOptions
+        options.from = _walletAddress
+        options.gasPrice = .automatic
+        options.gasLimit = .automatic
+        let method = "balanceOf"
+        guard let tx = contract.read(
```
Comment (tienpx3010) [inline CryptoWallet/Repositories/EthereumInteraction.swift]: Viết rõ nghĩa tx

**Comment by vanpd-0859** [comment]
ready

**Review by tienpx3010 — APPROVED** [review]

**Pull request merged**